ios/RNSScreenStackHeaderConfig.mm

```diff
@@ -670,6 +678,17 @@ + (void)updateViewController:(UIViewController *)vc
 
     UINavigationBarAppearance *scrollEdgeAppearance =
         [[UINavigationBarAppearance alloc] initWithBarAppearance:appearance];
+    if (config.largeTitleBackgroundColor != nil) {
+      // Add support for using a fully transparent bar when the backgroundColor is set to transparent. 
+      if (CGColorGetAlpha(config.largeTitleBackgroundColor.CGColor) == 0.) {
+      // This will also remove the background blur effect in the large title which is otherwise inherited from the standard appearance.
+        [scrollEdgeAppearance configureWithTransparentBackground];
+        // This must be set to nil otherwise a default view will be added to the navigation bar background with an opaque background.
+        scrollEdgeAppearance.backgroundColor = nil;
+      } else {
+        scrollEdgeAppearance.backgroundColor = config.largeTitleBackgroundColor;
+      }
+    }
     if (config.largeTitleBackgroundColor != nil) {
       scrollEdgeAppearance.backgroundColor = config.largeTitleBackgroundColor;
     }
```

Comment:
It looks like something is off here. You now have two "if" statements checking exactly for the same condition `if (config.largeTitleBackgroundColor != nil)`, while in first one you set `scrolleEdgeAppearance.backgroundColor` to nonnull value only when alpha is not 0, and then you override the background color no matter the alpha. Are you sure this works as you expected?